src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -104,30 +81,54 @@ class JesBackend extends Backend with LazyLogging {
     }
   }
 
+  sealed trait JesParameter {
+    def name: String
+    def gcs: String
+    def local: Path
+    def parameterType: String
+
+    final val isInput = this.isInstanceOf[JesInput]
```

Comment:
Not quite as you really should explicitly mention all instances of the sealed trait. But that's the general idea